tests/framework/extensions/configmaps/configmaps.go

```diff
@@ -0,0 +1,66 @@
+package configmaps
+
+import (
+	"context"
+
+	"github.com/rancher/rancher/pkg/api/scheme"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/unstructured"
+	coreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var ConfigMapGroupVersionResource = schema.GroupVersionResource{
+	Group:    "",
+	Version:  "v1",
+	Resource: "configmaps",
+}
+
+// CreateConfigMap is a helper function that uses the dynamic client to create a config map on a namespace for a specific cluster.
+// It registers a delete fuction.
+func CreateConfigMap(client *rancher.Client, clusterName, configMapName, description, namespace string, data, labels, annotations map[string]string) (*coreV1.ConfigMap, error) {
+	// ConfigMap object for a namespace in a cluster
+	annotations["field.cattle.io/description"] = description
+	configMap := &coreV1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
This is redundant, the k8s client will derive this from the struct type.

Reply:
Do you mean doing the TypeMeta is redundant?